cpp/src/strings/convert/convert_integers.cu

```diff
@@ -179,7 +179,9 @@ struct integer_to_string_size_fn
         if( value==0 )
             return 1;
         bool is_negative = value < 0;
-        value = abs(value);
+        // abs(std::numeric_limits<IntegerType>::min()) is negative
+        value = (value == std::numeric_limits<IntegerType>::min()) ?
```

Comment:
It feels weird to me pretending that `abs(min()) == max()` even though it should result in the same number of digits. This does seem like a clean way to handle generic `IntegerType` though. I guess alternatively, you could have a separate branch to compute digits for negative values? Just a comment, not a change request.

Reply:
Added more explanation to the comment.